component/src/main/java/org/wso2/extension/siddhi/execution/env/GetOriginIPFromXForwardedFunction.java

```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * The initialization method for GetOriginIPFromXForwardedFunction,
+     * this method will be called before the other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameter.
+     * @param siddhiAppContext             the context of the execution plan.
+     */
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ConfigReader reader,
+                        SiddhiAppContext siddhiAppContext) {
+        if (attributeExpressionExecutors.length != 1) {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to env:getOriginIPFromXForwarded" +
+                    "() function, required 1, but found " + attributeExpressionExecutors.length);
+        }
+
+        Attribute.Type attributeType = attributeExpressionExecutors[0].getReturnType();
+        if (attributeType != Attribute.Type.STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for first argument " +
+                    "'xForwardedHeader' of env:getOriginIPFromXForwarded() function, required " + Attribute.Type
+                    .STRING + ", but found " + attributeType.toString());
+        }
+
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are more than one function parameter.
+     *
+     * @param data the runtime values of function parameters.
+     * @return the function result.
+     */
+    @Override
+    protected Object execute(Object[] data) {
+        //we only allow single parameter for this function
+        return null;
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are zero or one function parameter.
+     *
+     * @param data null if the function parameter count is zero or
+     *             runtime data value of the function parameter.
+     * @return the function result.
+     */
+    @Override
+    protected Object execute(Object data) {
+        String xForwardedFor = data.toString();
+        if (xForwardedFor.isEmpty()) {
+            return null;
+        }
+        String extractedIPs[] = xForwardedFor.split(",", -1);
+        Matcher privateIPAddressMatcher;
+        Matcher ipAddressMatcher;
+        String filteredPublicIp = null;
+        String extractedIP;
+        for (int i = 0, extractedIPsLength = extractedIPs.length; i < extractedIPsLength; i++) {
+            extractedIP = extractedIPs[i].trim();
+            privateIPAddressMatcher = privateIPPAddressPattern.matcher(extractedIP);
+            ipAddressMatcher = ipPAddressPattern.matcher(extractedIP);
+            if (!ipAddressMatcher.matches()) {
+                continue;
+            }
+            if (!privateIPAddressMatcher.matches()) {
+                filteredPublicIp = extractedIP;
+                break;
+            }
+        }
```

Comment:
can we do the check like? because I assume always ipAddressMatcher can pass so let's use the less probable check first so that we can reduce the number of regex check and matcher creation.  

privateIPAddressMatcher = privateIPPAddressPattern.matcher(extractedIP);
+            if (!privateIPAddressMatcher.matches()) {
+                  ipAddressMatcher = ipPAddressPattern.matcher(extractedIP);
+                  if (ipAddressMatcher.matches()) {
+                       filteredPublicIp = extractedIP;
+                       break;
+                  }

+            }